iidm/iidm-modification/src/main/java/com/powsybl/iidm/modification/ReplaceThreeWindingsTransformersBy3TwoWindingsTransformers.java

```diff
@@ -0,0 +1,352 @@
+        regulatedTerminalMapping.put(t3w.getLeg1().getTerminal(), t2w1.getTerminal1());
+        regulatedTerminalMapping.put(t3w.getLeg2().getTerminal(), t2w2.getTerminal1());
+        regulatedTerminalMapping.put(t3w.getLeg3().getTerminal(), t2w3.getTerminal1());
+        return regulatedTerminalMapping;
+    }
+
+    private static void replaceRegulatedTerminal(TwoWindingsTransformer t2w, Map<Terminal, Terminal> regulatedTerminalMapping) {
+        t2w.getOptionalRatioTapChanger().ifPresent(rtc -> {
+            if (regulatedTerminalMapping.containsKey(rtc.getRegulationTerminal())) {
+                rtc.setRegulationTerminal(regulatedTerminalMapping.get(rtc.getRegulationTerminal()));
+            }
+        });
+        t2w.getOptionalPhaseTapChanger().ifPresent(ptc -> {
+            if (regulatedTerminalMapping.containsKey(ptc.getRegulationTerminal())) {
+                ptc.setRegulationTerminal(regulatedTerminalMapping.get(ptc.getRegulationTerminal()));
+            }
+        });
+    }
+
+    private static List<String> copyProperties(ThreeWindingsTransformer t3w, TwoWindingsTransformer t2wLeg1, TwoWindingsTransformer t2wLeg2, TwoWindingsTransformer t2wLeg3, VoltageLevel starVoltageLevel) {
+        List<String> lostProperties = new ArrayList<>();
+        t3w.getPropertyNames().forEach(propertyName -> {
+            boolean copied = copyProperty(propertyName, t3w.getProperty(propertyName), t2wLeg1, t2wLeg2, t2wLeg3, starVoltageLevel);
+            if (!copied) {
+                lostProperties.add(propertyName);
+            }
+        });
+        return lostProperties;
+    }
+
+    private static boolean copyProperty(String propertyName, String property, TwoWindingsTransformer t2wLeg1, TwoWindingsTransformer t2wLeg2, TwoWindingsTransformer t2wLeg3, VoltageLevel starVoltageLevel) {
+        boolean copied = true;
+        if (propertyName.equals("v")) {
+            starVoltageLevel.getBusView().getBuses().iterator().next().setV(Double.parseDouble(property));
+        } else if (propertyName.equals("angle")) {
+            starVoltageLevel.getBusView().getBuses().iterator().next().setAngle(Double.parseDouble(property));
+        } else if (propertyName.startsWith("CGMES.OperationalLimitSet_")) {
+            if (t2wLeg1.getOperationalLimitsGroups1().stream().anyMatch(operationalLimitsGroup -> propertyName.contains(operationalLimitsGroup.getId()))) {
+                t2wLeg1.setProperty(propertyName, property);
+            } else if (t2wLeg2.getOperationalLimitsGroups1().stream().anyMatch(operationalLimitsGroup -> propertyName.contains(operationalLimitsGroup.getId()))) {
+                t2wLeg2.setProperty(propertyName, property);
+            } else if (t2wLeg3.getOperationalLimitsGroups1().stream().anyMatch(operationalLimitsGroup -> propertyName.contains(operationalLimitsGroup.getId()))) {
+                t2wLeg3.setProperty(propertyName, property);
+            } else {
+                copied = false;
```

Comment:
Properties are used to pass additional data on an identifiable. They may be used in a full IIDM context (without exporting the network in another format) and it can make sense to transfer them.

Reply:
Done. All other properties  have been copied on the 3 2wt